src/CommonDBRelation.php

```diff
@@ -2126,7 +2126,12 @@ public static function countForMainItem(CommonDBTM $item, $extra_types_where = [
                 continue;
             }
 
-            $params = static::getTypeItemsQueryParams($item->fields['id'], $data['itemtype']);
+            $linked_item = getItemForItemtype($data['itemtype']);
+            $params = static::getTypeItemsQueryParams(
+                $item->fields['id'],
+                $data['itemtype'],
+                $linked_item->isEntityAssign(),
```

Comment:
It means that you remove the entity restrict criteria if the item can be assigned to entities. It does not seems logical to me.

Reply:
Indeed, it seems the real issue was that the answer table does in fact support entities but does not support recursion.